token/core/zkatdlog/crypto/rp/bulletproof_test.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package rp_test
+
+import (
+	math2 "math"
+	"strconv"
+
+	math "github.com/IBM/mathlib"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/core/zkatdlog/crypto/rp"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Range Proof", func() {
+
+	Describe("IPA", func() {
+		Context("If IPA is generated correctly", func() {
+			It("Succeeds", func() {
+				curve := math.Curves[1]
+				nr := 3
+				l := int(math2.Pow(2, float64(nr)))
+				leftGens := make([]*math.G1, l)
+				rightGens := make([]*math.G1, l)
+
+				rand, err := curve.Rand()
+				Expect(err).NotTo(HaveOccurred())
+
```

Comment:
why are they all the same?

Reply:
Not clear what this comment refers to?

Reply:
you selected all points Q,P,H,G with the same exponent